js/ui/components/clubhouse.js

```diff
@@ -628,16 +631,31 @@ var ClubhouseOpenButton = new Lang.Class({
 
     _init: function(params) {
         params = params || {};
-        this._icon = new St.Icon({ style_class: 'clubhouse-open-button-icon' });
-        params.child = this._icon;
+        let gfile =
+            Gio.File.new_for_uri('resource:///org/gnome/shell/theme/clubhouse-icon-pulse.png');
+        this._pulseAnimation = new Animation(gfile,
+                                             CLUBHOUSE_BUTTON_SIZE,
+                                             CLUBHOUSE_BUTTON_SIZE,
+                                             CLUBHOUSE_BUTTON_PULSE_SPEED);
+        this._pulseIcon = this._pulseAnimation.actor;
+
+        this._normalIcon = new St.Icon({ style_class: 'clubhouse-open-button-icon',
+                                         width: CLUBHOUSE_BUTTON_SIZE,
+                                         height: CLUBHOUSE_BUTTON_SIZE
+                                       });
+
+        params.child = this._normalIcon;
         this.parent(params);
     },
 
     setHighlighted: function(highlighted) {
-        if (highlighted)
-            this._icon.add_style_pseudo_class('highlighted');
-        else
-            this._icon.remove_style_pseudo_class('highlighted');
+        if (highlighted) {
+            this.child = this._pulseIcon;
+            this._pulseAnimation.play();
+        } else {
+            this.child = this._normalIcon;
+            this._pulseAnimation.stop();
```

Comment:
I have tried to use a bin layout to layer both icons and just affect their visible property in this function, but it was not working (due to the way the base class for the button works AFAIU).